src/ConfigFile.cpp

```diff
@@ -112,6 +118,12 @@ ConfigFile createDummyConfig()
 	serverConfig8081.port = "8081";
 	serverConfig8081.serverName = "cgi";
 
+	ConfigServer serverConfig8082;
+	serverConfig8081.locations.push_back(location7);
+	serverConfig8081.host = "127.0.0.1";
+	serverConfig8081.port = "8082";
+	serverConfig8081.serverName = "post";
```

Comment:
Only with the extra server we were able to discover the segfault occuring when no location was matched (which as such should not happen anymore with the default location, but nonetheless it would be well to implement the throw in the TargetResourceHandler as discussed -> we should also open an issue for that). So would keep it for the moment. Fixed the serverConfig name. I guess we are going to get rid of all these configs at the end anyways.

Reply:
But there already is another server which has only one location: 8081

I would not get rid of the config generation since it's a way to generate config for unittests (and isolate the tests from parsing). But it would make sense to extract the config generation into it's own file and move to test directory.

I'll open issue